judge/models/problem.py

```diff
@@ -535,3 +558,38 @@ class Meta:
         )
         verbose_name = _('solution')
         verbose_name_plural = _('solutions')
+
+
+class ProblemPointsVote(models.Model):
+    points = models.IntegerField(
+        verbose_name=_('proposed points'),
+        help_text=_('the amount of points the voter thinks this problem deserves.'),
+        validators=[
+            MinValueValidator(settings.DMOJ_PROBLEM_MIN_USER_POINTS_VOTE),
+            MaxValueValidator(settings.DMOJ_PROBLEM_MAX_USER_POINTS_VOTE),
+        ],
+    )
+
+    voter = models.ForeignKey(Profile, related_name='problem_points_votes', on_delete=CASCADE)
+    problem = models.ForeignKey(Problem, related_name='problem_points_votes', on_delete=CASCADE)
+    vote_time = models.DateTimeField(
+        verbose_name=_('vote time'),
+        help_text=_('the time this vote was cast'),
+        auto_now_add=True,
+        blank=True,
+    )
+
+    note = models.TextField(
+        verbose_name=_('note'),
+        help_text=_('justification for problem point value.'),
+        max_length=2048,
+        blank=True,
+        default='',
+    )
+
+    class Meta:
+        verbose_name = _('vote')
+        verbose_name_plural = _('votes')
```

Comment:
This shows up on the admin sidebar as "Votes". In following with "Contest Participations", this should probably be "problem votes".

Reply:
Or "problem point votes" or "problem points votes" but the latter is not grammatical I think...

Reply:
Ok will change.

Reply:
Wait, should votes be a subsection of problems like contest participations?

Reply:
Yes, that was my point in an earlier comment